lib/mutgen.c

```diff
@@ -25,6 +25,17 @@
 
 #include "msprime.h"
 
+typedef struct {
+    tsk_mutation_t *mutation;
+    struct mutation_node_t *next;
+} mutation_node_t;
+
+typedef struct {
+    tsk_site_t *site;
+    mutation_node_t *mutations_head;
```

Comment:
Maybe call this ``mutations`` ? (But again, mutations_head is strictly more correct I agree.)

Reply:
I want to keep this to distinguish it from the `mutations` property that a `site` has.